rust/examples/gscan/src/main.rs

```diff
@@ -58,59 +61,56 @@ async fn main() -> VResult<()> {
     let client_id = matches.get_one::<String>("client_id").unwrap();
     let client_secret = matches.get_one::<String>("client_secret").unwrap();
 
-    let token = Vaas::get_token(client_id, client_secret).await?;
+    let authenticator = ClientCredentials::new(client_id.to_owned(), client_secret.to_owned());
+    let vaas_connection = Vaas::builder(authenticator).build()?.connect().await?;
 
-    let file_verdicts = scan_files(&files, &token).await?;
-    let url_verdicts = scan_urls(&urls, &token).await?;
+    let file_verdicts = scan_files(&files, &vaas_connection).await?;
+    let url_verdicts = scan_urls(&urls, &vaas_connection).await?;
 
-    file_verdicts.iter().for_each(|(f, v)| {
-        println!(
-            "File: {:?} -> {}",
-            f,
-            match v {
-                Ok(v) => v.verdict.to_string(),
-                Err(e) => e.to_string(),
-            }
-        )
-    });
+    file_verdicts
+        .iter()
+        .for_each(|(f, v)| print_verdicts(f.display().to_string(), v));
 
-    url_verdicts.iter().for_each(|(u, v)| {
-        println!(
-            "Url: {:?} -> {}",
-            u.to_string(),
-            match v {
-                Ok(v) => v.verdict.to_string(),
-                Err(e) => e.to_string(),
-            }
-        )
-    });
+    url_verdicts.iter().for_each(|(u, v)| print_verdicts(u, v));
 
     Ok(())
 }
 
+fn print_verdicts<I: AsRef<str>>(i: I, v: &VResult<VaasVerdict>) {
+    print!("{} -> ", i.as_ref());
+    match v {
+        Ok(v) => {
+            print!("{}", v.verdict);
+            if let Some(detection) = &v.detection {
+                print!(" {}", detection);
+            }
+            println!();
+        }
+        Err(e) => {
+            println!("{}", e.to_string());

```

Comment:
As our error should implement `Display`, this should be sufficient:
```suggestion
            println!("{}", e);

```

Reply:
DONE